### ISSUE TITLE
Topology groups and PDB files

### ISSUE THREAD

**Issue by mimischi** [MDAnalysis/mdanalysis#1257]
Just stumbled upon the problem, that accessing any method that involves topology groups does not work, when creating a `Universe` with a PDB file. Both the  `master` version 0.15.0, as well as a freshly cloned `develop` version 0.16.0-dev do not work, while producing different errors.

It does work when loading the topology and trajectory files from the test suite, so I'm using that as the expected behaviour. An non-working example PDB file would be [1UBQ from PDB](http://www.rcsb.org/pdb/explore.do?structureId=1UBQ).

### Expected behaviour

```
u = mda.Universe(PSF, DCD)
u.dihedrals
<TopologyGroup containing 8921 Dihedrals>
```

#### MDAnalysis 0.15.0

```
u = mda.Universe('1ubq.pdb')
u.dihedrals
<TopologyGroup containing 0 Nones>
```

#### MDAnalysis 0.16.0-dev

```
u = mda.Universe('1ubq.pdb')
u.dihedrals
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-45-133525c932b4> in <module>()
      1 c = mda.Universe('1ubq.pdb')
----> 2 c.dihedrals

~/Code/mdanalysis/package/MDAnalysis/core/universe.pyc in dihedrals(self)
    516     def dihedrals(self):
    517         """Dihedral angles between atoms"""
--> 518         return self.atoms.dihedrals
    519 
    520     @property

~/Code/mdanalysis/package/MDAnalysis/core/groups.pyc in __getattr__(self, attr)
   1503                 pass
   1504         raise AttributeError("{cls} has no attribute {attr}".format(
-> 1505             cls=self.__class__.__name__, attr=attr))
   1506 
   1507     def __reduce__(self):

AttributeError: AtomGroup has no attribute dihedrals
```

The `.angles` does not work either.

Looking through the issues I came across #1253, which sounds pretty familiar.

**Comment by richardjgowers** [comment]
Hey, thanks for the report

So a pdb/gro doesn't technically define dihedrals or angles, but obviously they can be inferred from bonds. So in a way this is "working as intended".

There's currently `u.atoms.guess_bonds()` which will try to guess bonds (based on distances and guessed vdw radii), then angles, then dihedrals.  If you call this explicitly you should get the bonds from your PDB file.  (Assuming the vdw radius can be guessed for all atoms). 

If you load a PDB with CONECT records, then you can't immediately use them to guess angles admittedly, maybe there should be a `u.atoms.guess_angles()` method too..

**Comment by mimischi** [comment]
I see, you have to call `u.atoms.guess_bonds()` explicitly or set `guess_bonds=True` for the Universe object. Thanks!

Wouldn't it be beneficial to catch that exception in the current `develop` version?

**Comment by richardjgowers** [comment]
Yeah the error message could definitely be better, like providing a customised AttributeError for a known list of things

**Comment by mimischi** [comment]
I'd be happy to contribute if that's ok!

**Comment by richardjgowers** [comment]
@mimischi yeah that would be great!  I think if you extended the `__getattr__` [here](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L1491) to add a special case for `bonds` etc.  If you're doing that, you should also [write a test that checks the errormessage](https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/core/test_group_traj_access.py#L48)

**Comment by mimischi** [comment]
Finally coming back to this issue I noticed the depreciation warnings and #1377.

Does the issue now shift from [`AtomGroup.__getattr__`](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L1522) to [`Atom.__getattr__`](https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/core/groups.py#L2533)?

**Comment by mimischi** [comment]
Ping 👋

**Comment by kain88-de** [comment]
ping @richardjgowers can you comment on this?
